pkg/controller/logstorage/linseed/linseed_controller.go

```diff
@@ -0,0 +1,389 @@
+		return reconcile.Result{}, nil
+	}
+
+	// Create a trusted bundle to pass to the render pacakge. The actual contents of this bundle don't matter - the ConfigMap
+	// itself will be managed by the Secret controller. But, we need an interface to use as an argument to render in order
+	// to configure volume mounts properly.
+	trustedBundle := certificatemanagement.CreateTrustedBundle()
+
+	var esClusterConfig *relasticsearch.ClusterConfig
+	if managementClusterConnection == nil {
+		flowShards := logstoragecommon.CalculateFlowShards(args.LogStorage.Spec.Nodes, logstoragecommon.DefaultElasticsearchShards)
+		esClusterConfig = relasticsearch.NewClusterConfig(render.DefaultElasticsearchClusterName, args.LogStorage.Replicas(), logstoragecommon.DefaultElasticsearchShards, flowShards)
+	}
+
+	// Determine the namespaces to which we must bind the linseed cluster role.
+	bindNamespaces := []string{request.InstallNamespace()}
+	if r.multiTenant {
+		bindNamespaces, err = utils.TenantNamespaces(ctx, r.client)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	cfg := &linseed.Config{
+		Installation:      install,
+		PullSecrets:       pullSecrets,
+		Namespace:         request.InstallNamespace(),
+		BindNamespaces:    bindNamespaces,
+		TrustedBundle:     trustedBundle,
+		ClusterDomain:     r.clusterDomain,
+		KeyPair:           linseedKeyPair,
+		TokenKeyPair:      tokenKeyPair,
+		UsePSP:            r.usePSP,
+		ESClusterConfig:   esClusterConfig,
+		ManagementCluster: managementCluster != nil,
+		Tenant:            args.Tenant,
+	}
+	linseedComponent := linseed.Linseed(cfg)
+
+	if err := imageset.ApplyImageSet(ctx, r.client, variant, linseedComponent); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// In standard installs, the LogStorage owns Linseed. For multi-tenant, it's owned by the Manager instance.
+	var hdler utils.ComponentHandler
+	if r.multiTenant {
+		hdler = utils.NewComponentHandler(reqLogger, r.client, r.scheme, args.Tenant)
+	} else {
+		hdler = utils.NewComponentHandler(reqLogger, r.client, r.scheme, args.LogStorage)
+	}
+	for _, comp := range []render.Component{linseedComponent} {
+		if err := hdler.CreateOrUpdateOrDelete(ctx, comp, r.status); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating / deleting resource", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// TODO: For now, just do ESGW here since it serves a similar purpose to Linseed.
+	if err := r.createESGateway(
+		ctx,
+		request,
+		install,
+		variant,
+		pullSecrets,
+		hdler,
+		reqLogger,
+		r.usePSP,
+	); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// TODO: This is a temporary location for this. Ideally this goes in its own controller, as it likely
+	// needs to be multi-tenant!
```

Comment:
Update: ideally this isn't needed at all for multi-tenant clusters so it can just go into the single-tenant "elastic" controller or somewhere else.